src/main/java/edu/ksu/canvas/attendance/controller/CourseConfigurationController.java

```diff
@@ -93,6 +119,71 @@ public ModelAndView synchronizeWithCanvas(@PathVariable String sectionId) throws
+            bindingResult.getFieldErrors().forEach(error -> errors.add(error.getCode()));
+            page.addObject("error", errors);
+            page.addObject("selectedSectionId", sectionId);
+            return page;
+        } else {
+            LOG.info("eid: " + canvasService.getEid() + " is pushing grades for course # " + canvasService.getCourseId() + " to Canvas");
+            ModelAndView page = new ModelAndView("forward:/courseConfiguration/" + sectionId);
+
+            Long courseId = Long.valueOf(canvasService.getCourseId());
+
+            boolean isSimpleAttendance = classSetupForm.getSimpleAttendance();
+            List<AttendanceSummaryModel> summaryForSections = isSimpleAttendance ?
+                    reportService.getSimpleAttendanceSummaryReport(sectionId) :
+                    reportService.getAviationAttendanceSummaryReport(sectionId);
+
+            AttendanceAssignment assignmentConfigurationFromSetup = generateAssignmentFromClassSetupForm(classSetupForm);
+            try {
+                assignmentSubmitter.submitCourseAttendances(isSimpleAttendance, summaryForSections, courseId, canvasService.getOauthToken(), assignmentConfigurationFromSetup);
+                page.addObject("pushingSuccessful", true);
+            }
+            catch (AttendanceAssignmentException e){
+                page.addObject("error", e.getMessage());
+            }
+
+
+
+            return page;
+        }
+    }
+
+    private AttendanceAssignment generateAssignmentFromClassSetupForm(CourseConfigurationForm classSetupForm) {
+        AttendanceAssignment assignmentConfigurationFromSetup = new AttendanceAssignment();
+        assignmentConfigurationFromSetup.setAssignmentPoints(classSetupForm.getAssignmentPoints());
+        assignmentConfigurationFromSetup.setGradingOn(true);
+        assignmentConfigurationFromSetup.setPresentPoints(classSetupForm.getPresentPoints());
+        assignmentConfigurationFromSetup.setTardyPoints(classSetupForm.getTardyPoints());
+        assignmentConfigurationFromSetup.setExcusedPoints(classSetupForm.getExcusedPoints());
+        assignmentConfigurationFromSetup.setAbsentPoints(classSetupForm.getAbsentPoints());
+        assignmentConfigurationFromSetup.setAssignmentName(classSetupForm.getAssignmentName());
+        return assignmentConfigurationFromSetup;
+    }
 
+    @RequestMapping(value = "/{sectionId}/delete", method = RequestMethod.POST)
+    public ModelAndView deleteAttendanceAssignment(@PathVariable String sectionId) throws NoLtiSessionException {
+        LOG.info("eid: " + canvasService.getEid() + " is turning off grading feature and deleting the assignment from Canvas for section: " + sectionId);
+        ModelAndView page = new ModelAndView("forward:/courseConfiguration/" + sectionId);
+
+        try {
+
+            assignmentAssistant.deleteAssignmentInCanvas(canvasService.getCourseId().longValue(), canvasService.getOauthToken());
+        } catch (Exception exception) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1166)